rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -138,12 +140,12 @@ module Impl {
   }
 
   /** A path expression that may access a local variable. */
-  private class VariableAccessCand extends PathExpr {
+  private class VariableAccessCand extends TVariableAccess {
```

Comment:
I wonder if it would be possible to implement this without special cases that need handling in `VariableImpl.qll`? Perhaps if format strings with variable accesses gave rise to a normal `PathExpr` under the `FormatArgsExpr` node? Then they could be picked up by `VariableAccessCand` like any other variable access.

Reply:
To generate a "real" `PathExpr` we'd need to change the extractor.

Reply:
we could theoretically:

* insert a `GenericPathExpr` superclass of `PathExpr` (via `@annotate(PathExpr, replace_bases={Expr: GenericPathExpr})`), maybe taking over all fields from it
* add a `FormatPathExpr` synth node beneath it
* replace `PathExpr` with `GenericPathExpr` in current `VariableImpl`